decode.go

```diff
@@ -66,9 +66,10 @@ func (e *DataReconstructionError) Error() string {
 }
 
 // DecodeSnapshot restores an entire snapshot to dst.
-func DecodeSnapshot(repository Repository, snapshot *Snapshot, dst string, excludes []string, pedantic bool) (chan Progress, error) {
+func DecodeSnapshot(repository Repository, snapshot *Snapshot, dst string, excludes []string, pedantic bool) (<-chan Progress, error) {
 	prog := make(chan Progress)
-	go func() {
+	go func(prog chan<- Progress) {
```

Comment:
What are the advantages for passing the channel to the go routines here and below?

Reply:
Making clear that they are closed within this goroutine.

Reply:
since its not done this way by golang [examples](https://blog.golang.org/pipelines) itself, i removed the parameter again